src/lib/services/access-service.ts

```diff
@@ -0,0 +1,219 @@
+        const role = roles.find(r => r.type === 'root' && r.name === roleName);
+        if(role) {
+            try {
+                await Promise.all(currentRootRoles.map(r => this.store.removeUserFromRole(userId, r.id)));
+                await this.store.addUserToRole(userId, role.id);
+            } catch (error) {
+                this.logger.warn('Could not add role=${roleName} to userId=${userId}');
+            }
+        }
+    }
+
+    async removeUserFromRole(userId: number, roleId: number) {
+        return this.store.removeUserFromRole(userId, roleId);
+    }
+
+    async addPermissionToRole(roleId: number, permission: string, projectId?: string) {
+        if(isProjectPermission(permission) && !projectId) {
+            throw new Error(`ProjectId cannot be empty for permission=${permission}`)
+        } 
+        return this.store.addPermissionsToRole(roleId, [permission], projectId);
+    }
+
+    async removePermissionFromRole(roleId: number, permission: string, projectId?: string) {
+        if(isProjectPermission(permission) && !projectId) {
+            throw new Error(`ProjectId cannot be empty for permission=${permission}`)
+        }
+        return this.store.removePermissionFromRole(roleId, permission, projectId);
+    }
+
+    async getRoles(): Promise<Role[]> {
+        return this.store.getRoles();
+    }
+
+    async getRole(roleId: number): Promise<RoleData> {
+        const [role, permissions, users] = await Promise.all([
+            this.store.getRoleWithId(roleId),
+            this.store.getPermissionsForRole(roleId),
+            this.getUsersForRole(roleId),
+        ]);
+        return { role, permissions, users };
+    }
+
+    async getRolesForProject(projectId: string): Promise<Role[]> {
+        return this.store.getRolesForProject(projectId);
+    }
+
+    async getRolesForUser(userId: number): Promise<Role[]> {
+        return this.store.getRolesForUserId(userId);
+    }
+
+    async getUsersForRole(roleId) : Promise<User[]> {
+        const userIdList = await this.store.getUserIdsForRole(roleId);
+        return this.userStore.getAllWithId(userIdList);
+    }
+
+    // Move to project-service?
+    async getProjectRoleUsers(projectId: string): Promise<[Role[], UserWithRole[]]> {
+        const roles = await this.store.getRolesForProject(projectId);
+
+        const users = await Promise.all(roles.map(async role => {
+            const users = await this.getUsersForRole(role.id);
+            return users.map(u => ({ ...u, roleId: role.id }))
+        }));
+        return [roles, users.flat()];
+    }
+
+    async createDefaultProjectRoles(owner: User, projectId: string) {
+        if(!projectId) {
+            throw new Error("ProjectId cannot be empty");
+        }
+
+        const adminRole = await this.store.createRole(
+            RoleName.ADMIN,
+            RoleType.PROJECT,
+            projectId,
+            `Admin role for project "${projectId}"`,
```

Comment:
Do these override the default descriptions in the migration?

Reply:
Description suggestions: 

Project Admin
Users with the project admin role have full control over the project, and can add and manage other users within the project context, manage feature toggles within the project, and control advanced project features like archiving and deleting the project.

Project Regular
Users with the regular role within a project are allowed to view, create and update feature toggles, but have limited permissions in regards to managing the projects user access and can not archive or delete the project. 

Global Admin: 
Users with the global admin role have superuser access to Unleash and can perform any operation within the unleash platform.

Global Regular:
Users with the global regular role have access most features in Unleash, but can not manage users and roles in the global scope. If a user with a global regular role creates a project, they will become a project admin and receive superuser rights within the context of that project.

Reply:
The migration only takes care of the default "root roles". The project specific ones are created when a new project is created.